gdml-generator/src/SimpleLZ.cc

```diff
@@ -0,0 +1,231 @@
+       329.630743,  570.930392,  659.264065,  570.949257,  329.633322,
+       -47.940493,  -143.001204, -235.615124, -324.197605, -407.232986,
+       -483.300486, -551.098581, -609.467226, -657.407719, -694.099784,
+       -718.91561,  -731.43059,  -731.43059,  -718.91561,  -694.099784,
+       -657.407719, -609.467226, -551.098581, -483.300486, -407.232986,
+       -324.197605, -235.615124, -143.001204, -47.940493,  47.940493,
+       143.001204,  235.615124,  324.197605,  407.232986,  483.300486,
+       551.098581,  609.467226,  657.407719,  694.099784,  718.91561,
+       731.43059,   731.43059,   718.91561,   694.099784,  657.407719,
+       609.467226,  551.098581,  483.300486,  407.232986,  324.197605,
+       235.615124,  143.001204,  47.940493};
+}  // namespace
+
+//---------------------------------------------------------------------------//
+/*!
+ * Constructor
+ */
+SimpleLZ::SimpleLZ(int sqrt_num_pmts) : sqrt_num_pmts_(sqrt_num_pmts) {}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Build a simple LZ geometry, or a notional geometry with a set number of
+ * PMTs.
+ *
+ * This methods operates in two modes. If sqrt_num_pmts_ is zero (the default
+ * value) a simplified model of the LZ top PMT detector array is created with
+ * 253 PMTs placed in their real locations. If sqrt_num_pmts_ is positive, then
+ * a square-pitch sqrt_num_pnts_ by sqrt_num_pmts_ array of detectors created.
+ * The PMTs are spaced 1 mm apart, and sized such that the array is inscribed
+ * within the world cyclinder. These notional geometries are useful for
+ * performance studies.
+ */
+G4VPhysicalVolume* SimpleLZ::Construct()
+{
+    // Materials
+    auto nist = G4NistManager::Instance();
+    auto pmt_mat = nist->FindOrBuildMaterial("G4_Au");
+    auto world_mat = nist->FindOrBuildMaterial("G4_Galactic");
+    world_mat->SetName("vacuum");
+
+    // World Cylinder
+    G4Tubs* world_cylinder = new G4Tubs(
+        "world_cylinder", 0, 780.288 * mm, 123 * mm, 0 * deg, 360 * deg);
+    auto const world_lv
+        = new G4LogicalVolume(world_cylinder, world_mat, "world");
+    auto const world_pv = new G4PVPlacement(
+        nullptr, G4ThreeVector(), world_lv, "world_pv", nullptr, false, 0, false);
+
+    if (sqrt_num_pmts_ == 0)
+    {
+        // Simple LZ model
+        for (auto i = 0; i < 253; ++i)
+        {
+            G4Tubs* pmt
+                = new G4Tubs("pmt", 0, 39 * mm, 123 * mm, 0 * deg, 360 * deg);
+            std::string pmt_name = "pmt_" + std::to_string(i);
+            auto const pmt_lv = new G4LogicalVolume(pmt, pmt_mat, pmt_name);
+            new G4PVPlacement(nullptr,
+                              G4ThreeVector({pmt_x[i] * mm, pmt_y[i] * mm, 0}),
+                              pmt_lv,
+                              "pmt_pv",
+                              world_lv,
+                              false,
+                              0,
```

Comment:
I'd suggest making this copy number `i` instead of `0`. Not sure if it is relevant to your problem, but could be useful in the future, since you're placing the same logical volume multiple times. So when we score hits, we know which pmt is in the code by referring to its copy number, which is obviously directly mapped to its (x, y) position in the pmt position arrays.